Sources/MessagingPush/UserNotificationsFramework/UserNotificationsFrameworkAdapter.swift

```diff
@@ -22,19 +23,37 @@ protocol UserNotificationsFrameworkAdapter {
 /**
  Keep this class small and simple because it is only able to be tested in QA testing. All logic for handling push events should be in the rest of the code base that has automated tests around it.
  */
-// sourcery: InjectRegister = "UserNotificationsFrameworkAdapter"
 @available(iOSApplicationExtension, unavailable)
 class UserNotificationsFrameworkAdapterImpl: NSObject, UNUserNotificationCenterDelegate, UserNotificationsFrameworkAdapter {
-    private let pushEventHandler: PushEventHandler
-    private var userNotificationCenter: UserNotificationCenter
+    /*
+     # Why is this class not stored in the digraph?
 
-    private var notificationCenterDelegateProxy: PushEventHandlerProxy {
-        PushEventHandlerProxyImpl.shared
+     This class is the SDK's `UNUserNotificationCenterDelegate` instance.  Meaning, the CIO SDK registers this class with the OS in order to
+     receive callbacks when push notifications are interacted with.
+
+     It's important that the instance of this class provided to the OS stays in memory so our SDK can receive those OS callbacks.
+
+     In order to promise this class's singleton instance stays in memory, the singleton instance is *not* stored inside of the digraph (like all other singletons in our SDK is).
+     */
+    static let shared = UserNotificationsFrameworkAdapterImpl()
+
+    // It's important that we use do not keep a strong reference to any dependencies or to the digraph. Otherwise, the SDK would crash if the SDK's digraph gets re-initialized and
```

Comment:
That is a good call out but it is still something that if not managed correctly or by mistake/negligence can cause unexpected results. Can we avoid it? as in not using `PushEventHandler` at all for `DiGraph` just relying on stub/actual instance even for testing.

Reply:
Ah, I see your point, unfortunately with the current `digraph` implementation doesn't seem possible but I can easily someone holding a reference here by mistake. Another could be to take all these classes out of `digraph` but that would mean way more changes. 

Can we please a ticket about this so we don't lose sight of it? and we can probabaly try to create a way in our Sourcery template to have some objects as `App lifecycle` available.

Reply:
[ticket created](https://linear.app/customerio/issue/MBL-73/di-graph-gets-re-created-while-sdk-in-memory).